### PR TITLE
added more books and copies api and tests

### PR DESCRIPTION
**Pull request by RiTiS7**

### CODE REVIEW

File: server/controllers/books.js [hunk 1]
```diff
@@ -31,65 +36,114 @@ module.exports = {
...
+    if (!book) {
+      return res.status(404).json({ message: "Can't delete copy, book was not found" });
+    }
+
+    const removedCopy = await Copy.findOneAndRemove({ ID: copyID });
+    if (!removedCopy) {
+      return res.status(404).json({ message: "Can't delete copy, copy was not found" });
+    }
+
+    const copyIndex = book.copies.indexOf(removedCopy._id);
```
Comment (Alaev) [inline server/controllers/books.js]: lodash
https://lodash.com/docs/4.17.4#indexOf

File: server/controllers/books.js [hunk 2]
```diff
@@ -31,65 +36,114 @@ module.exports = {
...
+    if (copyIndex === -1) {
+      return res.status(404).json({ message: "Can't delete copy from book, copy was not found" });
+    }
+
+    book.copies.splice(copyIndex, 1);
```
Comment (Alaev) [inline server/controllers/books.js]: lodash
https://lodash.com/docs/4.17.4#without

File: server/routes/api/books.js [hunk 3]
```diff
@@ -3,23 +3,18 @@ const booksController = require('../../controllers/books');
...
   .route('/:bookISBN')
   .get(booksController.getBook)
   .put(booksController.replaceBook)
-  // .patch(booksController.updateBook)
   .delete(booksController.deleteBook);
 
-// router.route('/:bookISBN/copy')
-//   .get(booksController.getBookCopies)
-//   .post(booksController.newBookCopy);
+router.route('/:bookISBN/copy').get(booksController.getBookCopies).post(booksController.newBookCopy);
```
Comment (Alaev) [inline server/routes/api/books.js]: shouldn't the copy be as a parameter?
:copy and not only copy?

File: server/controllers/books.js [hunk 4]
```diff
@@ -1,4 +1,9 @@
+/* eslint no-underscore-dangle: ["error", { "allow": ["_id"] }]*/
+
+const _ = require('lodash');
 const Book = require('../models/Books');
+const Copy = require('../models/Copies');
+require('../models/Borrowers');
```
Comment (Alaev) [inline server/controllers/books.js]: why do we need it? do we use it here?

File: server/controllers/books.js [hunk 5]
```diff
@@ -1,4 +1,9 @@
+/* eslint no-underscore-dangle: ["error", { "allow": ["_id"] }]*/
+
+const _ = require('lodash');
```
Comment (Alaev) [inline server/controllers/books.js]: import only the functions that were used.

File: server/routes/api/books.js [hunk 6]
```diff
@@ -3,23 +3,18 @@ const booksController = require('../../controllers/books');
...
 
-// router.route('/:bookISBN/copy/:copyId')
-//   .get(booksController.getCopy)
-//   .put(booksController.updateBookCopy)
-//   .delete(booksController.deleteCopy);
+router
+  .route('/:bookISBN/copy/:copyID')
```
Comment (Alaev) [inline server/routes/api/books.js]: Is the route going to be /123/1/3432432432423
do we need to pass the copy id here? can't we use the copy param? as the copy should have unique id 
example:  ISBN 1214
                  copy 1214-1

File: server/controllers/books.js [hunk 7]
```diff
@@ -134,16 +137,19 @@ module.exports = {
...
-    const copyIndex = book.copies.indexOf(removedCopy._id);
+    // const copyInBook = _.includes(_.map(book.copies, v => v.toString()), _.toString(removedCopy._id));
+    const copyIndex= book.copies.indexOf(removedCopy._id);
     if (copyIndex === -1) {
+      new Copy(removedCopy).save();
       return res.status(404).json({ message: "Can't delete copy from book, copy was not found" });
     }
 
-    book.copies.splice(copyIndex, 1);
+    book.copies.remove(removedCopy._id);
```
Comment (Alaev) [inline server/controllers/books.js]: remove? is is a new function?
if you were trying to use lodash here you should have added _ 
also how dose the test pass? did you check for this scenario?

File: server/controllers/books.js [hunk 8]
```diff
@@ -62,7 +63,8 @@ module.exports = {
     const { bookISBN } = req.params;
     const book = await Book.findOne({ ISBN: bookISBN }, { _id: 0 }).populate({
       path: 'copies',
-      populate: { path: 'lastBorrower', select: 'name' }
+      select: '-_id',
+      populate: { path: 'lastBorrower', select: 'name -_id' }
```
Comment (Alaev) [inline server/controllers/books.js]: please add comment here why do you use -_id as it follows your code around

File: server/controllers/books.js [hunk 9]
```diff
@@ -134,16 +137,19 @@ module.exports = {
       return res.status(404).json({ message: "Can't delete copy, copy was not found" });
     }
 
-    const copyIndex = book.copies.indexOf(removedCopy._id);
+    // const copyInBook = _.includes(_.map(book.copies, v => v.toString()), _.toString(removedCopy._id));
+    const copyIndex= book.copies.indexOf(removedCopy._id);
```
Comment (Alaev) [inline server/controllers/books.js]: use lodash for indexof
https://lodash.com/docs/4.17.4#indexOf

File: server/controllers/books.js (hunk 8 above)
Comment (RiTiS7) [inline server/controllers/books.js]: we add the '-_id' so it won't display it in the output object of populate, just like in here:
Book.findOne({ ISBN: bookISBN }, '-_id -__v');

**Pull request merged**